app/renderer/ui/components/radonOperator/index.tsx

```diff
@@ -0,0 +1,184 @@
+import * as React from "react"
+
+import { getHashingFunctions, getMethodInformation, getMethodsByType, RadonType, getFilteringFunctions, getReducingFunctions } from "./radon"
+const styles = require("./style.scss")
+
+export interface RadonOperatorProps {
+  inputType: RadonType,
+}
+
+export interface RadonOperatorState {
+  outputType: RadonType,
+  availableMethodsNames: any,
+  selectedMethodInfo: any,
+  selectedFunctionFilter: any,
+  selectedFunctionReduce: any,
+  selectedFunctionHash: any,
+  functionFilterInput: any,
+  radonTypeInput: any,
+}
+
+/**
+ * RadonOperator ui component
+ *
+ * @export
+ * @class RadonOperator
+ * @extends {React.Component<RadonOperatorProps>}
+ */
+export default class RadonOperator extends React.Component<RadonOperatorProps, RadonOperatorState> {
+  constructor(props: RadonOperatorProps) {
+    super(props)
+
+    const availableMethodsNames = getMethodsByType(this.props.inputType)
+    const selectedMethodInfo = getMethodInformation(availableMethodsNames[0])
+
+    const defaultSelectedMethods = selectedMethodInfo.arguments.reduce((acc, item) => {
+      switch (item.kind) {
```

Comment:
Does this necessarily need to be a `switch` or could it be implemented declaratively as an object?
```ts
const defaultSelectedMethodsMap = {
  "FUNCTION_FILTER": getFilteringFunctions()[0],
 /* ... */
}

const defaultSelectedMethods = selectedMethodInfo
  .arguments
  .reduce((acc, item) => ({...acc, [item.kind]: defaultSelectedMethodsReducerMap[item.kind]}), {})
```

Reply:
Whatever. Don't take this as a recommendation for a change, I need yet to give a more detailed review to the entire PR to see if this makes sense altogether.

Reply:
This is a trick that I'm solving